### PR TITLE
SWC-6412 - non-feature related bugfixes

### PR DESCRIPTION
**Pull request by nickgros**
- Fix bug in UserSearchBoxV2 where the popover would close the search results menu
- Fix bug where the Popover transition could result in a thrown error if the Popover was shown within a Dialog
- Add thrown error from code review

### CODE REVIEW

File: packages/synapse-react-client/src/lib/containers/UserSearchBoxV2.tsx [hunk 1]
```diff
@@ -73,6 +73,7 @@ const customSelectComponents: Partial<
           userGroupHeader={data.header}
           disableHref={true}
           showFullName={true}
+          showCardOnHover={false}
```
Comment (nickgros) [inline packages/synapse-react-client/src/lib/containers/UserSearchBoxV2.tsx:76]: When the mouse hovered over the username in the search results, the search results would disappear!

File: packages/synapse-react-client/src/lib/utils/hooks/SynapseAPI/file/useFiles.ts [hunk 2]
```diff
@@ -93,9 +93,7 @@ export function useGetFileBatch(
   if (request.includePreSignedURLs || request.includePreviewPreSignedURLs) {
     // Don't use this hook if you need pre-signed URLs. They expire every 30 seconds, so you will either end up giving the
     // user an expired URL, or making requests to the backend more frequently than necessary.
-    console.error('useGetFileBatch does not support pre-signed URLs.')
-    request.includePreSignedURLs = false
-    request.includePreviewPreSignedURLs = false
+    throw new Error('useGetFileBatch does not support pre-signed URLs.')
```
Comment (nickgros) [inline packages/synapse-react-client/src/lib/utils/hooks/SynapseAPI/file/useFiles.ts:96]: From code review #168

File: packages/synapse-react-client/src/lib/utils/hooks/useOverlay.tsx [hunk 3]
```diff
@@ -78,6 +78,7 @@ export function useOverlay(
   const OverlayComponent = useCallback(
     () => (
       <Popover
+        TransitionComponent={Fade}
```
Comment (nickgros) [inline packages/synapse-react-client/src/lib/utils/hooks/useOverlay.tsx:81]: For some reason, the default `Grow` transition component would trigger an error if this Popover opened in a Dialog.

**Review by jay-hodgson — APPROVED** [review]

**Comment by nickgros** [comment]
Two different flaky tests failed, so I am merging this PR

**Pull request merged**